Core/include/Acts/EventData/SubspaceHelpers.hpp

```diff
@@ -0,0 +1,326 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/EventData/Types.hpp"
+#include "Acts/Utilities/AlgebraHelpers.hpp"
+#include "Acts/Utilities/Enumerate.hpp"
+
+#include <bitset>
+#include <span>
+
+#include <boost/container/static_vector.hpp>
+
+namespace Acts {
+
+/// Check subspace indices for consistency
+///
+/// @tparam Container type of the container
+///
+/// @param container the container of indices
+/// @param fullSize the full size of the subspace
+/// @param subspaceSize the size of the subspace
+///
+/// @return true if the indices are consistent
+template <typename Container>
+inline static bool checkSubspaceIndices(const Container& container,
+                                        std::size_t fullSize,
+                                        std::size_t subspaceSize) {
+  if (subspaceSize > fullSize) {
+    return false;
+  }
+  if (container.size() != subspaceSize) {
+    return false;
+  }
+  for (auto it = container.begin(); it != container.end();) {
+    auto index = *it;
+    if (index >= fullSize) {
+      return false;
+    }
+    ++it;
+    if (std::find(it, container.end(), index) != container.end()) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/// Serialize subspace indices to a single 64 bit integer
+///
+/// @tparam FullSize the full size of the subspace
+///
+/// @param indices the subspace indices
+///
+/// @return the serialized subspace indices
+template <std::size_t FullSize>
+inline static SerializedSubspaceIndices serializeSubspaceIndices(
```

Comment:
Now where I used the `Acts::MultiIndex` yesterday, I wonder if that would not be a good tool for implementing that...

Reply:
Could be but I don't really see a benefit in that. Do you have an example one how you think the implementation would get easier / more readable?

Reply:
Yeah I though it would rely on a already established and tested tool for such bit shifting buisness. But yeah not a strong preference.